src/lib/components/End/OnDate.js

```diff
@@ -15,32 +15,54 @@ const EndOnDate = ({
   },
   handleChange,
 }) => {
+  const CustomCalendar = options.calendarComponent;
+  const isCustomCalendarProvided = typeof CustomCalendar !== 'undefined';
```

Comment:
Do we need this check explicitly?
Are there any issues with using just `{CustomCalendar ? <CustomCalendar ... > : <DateTime>}`